File: 프로그래머스/2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for s in skill_trees:
        flag = False
        idx = 0
        for j in range(len(s)):
            # 안되는 경우
            if skill[idx] != s[j] and s[j] in skill:
                flag = True
                break
            # 맞는 경우
            if idx == len(skill)-1:
                break
            if skill[idx] == s[j]:
                idx +=1
        if not flag:
            answer +=1        
    return answer

File: 프로그래머스/2/49993. 스킬트리/스킬트리.js
function solution(skill, skill_trees) {
    let answer = 0;

    const regExp = new RegExp(`[^${skill}]`,"g");

    skill_trees.forEach(function(item) {
        if( skill.indexOf(item.replace(regExp,'')) === 0 ) {
            ++answer;
        }
    })

    return answer;
}